File: src/main/java/com/projetopicpay/springao/services/UsuarioService.java
package com.projetopicpay.springao.services;

import com.projetopicpay.springao.dtos.ErroResponse;
import com.projetopicpay.springao.dtos.UsuarioCadastroDTO;
import com.projetopicpay.springao.dtos.UsuarioResponseDTO;
import com.projetopicpay.springao.exception.UsuarioExistenteException;
import com.projetopicpay.springao.models.Usuario;
import com.projetopicpay.springao.repositories.UsuarioRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.util.Optional;


@Service
@RequiredArgsConstructor
public class UsuarioService {

    private final UsuarioRepository repository;

    public UsuarioResponseDTO criarUsuario(UsuarioCadastroDTO user){

        Optional<Usuario> usuario = repository.findByDocumentoOrEmail(user.documento(), user.email());

        if(usuario.isPresent()){
            throw new UsuarioExistenteException("usuario ja existente");
        }

        Usuario u = new Usuario();
        u.setDocumento(user.documento());
        u.setNome(user.nome());
        u.setSenha(user.senha());
        u.setEmail(user.email());
        u.setTipoCliente(user.tipoCliente());


        repository.save(u);


        return new UsuarioResponseDTO(u.getNome(), u.getEmail(), u.getDocumento(), u.getSaldo());
    }

}

File: src/services/usuarioService.ts
import bcrypt from 'bcryptjs';
import { prisma } from '../lib/prisma';
import { UsuarioCadastroDTO, UsuarioResponseDTO } from '../dtos/usuario.dto';
import { UsuarioExistenteException } from '../exceptions/AppError';

export async function criarUsuario(data: UsuarioCadastroDTO): Promise<UsuarioResponseDTO> {
  const existente = await prisma.usuario.findFirst({
    where: {
      OR: [{ documento: data.documento }, { email: data.email }],
    },
  });

  if (existente) {
    throw new UsuarioExistenteException();
  }

  const senhaHash = await bcrypt.hash(data.senha, 10);

  const usuario = await prisma.usuario.create({
    data: {
      nome: data.nome,
      email: data.email,
      senha: senhaHash,
      documento: data.documento,
      tipoCliente: data.tipoCliente,
    },
  });

  return {
    nome: usuario.nome,
    email: usuario.email,
    documento: usuario.documento,
    saldo: usuario.saldo.toString(),
  };
}
